### DISCUSSION TITLE
HTML Rendering Field Type for Collections

### DISCUSSION THREAD

**Discussion by RajamaniManoharan** [directus/directus#19694] [Feature Requests]
### Summary

The feature request proposed is to add a new field type to Directus, named "HTML Renderer", that would allow HTML content to be rendered and displayed directly within the Directus interface. This would provide a more convenient way for users to preview HTML content within Directus, without needing to use external tools or services. When this field type is selected, any HTML content within the field should be rendered and displayed in the Directus UI. This feature could be implemented as a Directus extension, similar to the custom displays that can be created using the Directus extensions SDK. It would need to safely parse and display HTML content while considering potential security implications such as cross-site scripting (XSS) attacks.

### Basic Example

```js
export default {
	id: 'html_renderer',
	name: 'HTML Renderer',
	icon: 'code',
	description: 'Renders HTML content.',
	component: function ({ value }) {
		// Safely parse and display HTML content
		return parseHTML(value);
	},
	options: null,
	types: ['string'],
};
```

### Motivation

This feature would provide a more convenient way for users to preview HTML content directly within Directus, without needing to rely on external tools or services. With this feature, users could manage and visualize their HTML content more effectively, leading to improved productivity and user experience.

### Detailed Design

Establish the Field Type: The HTML Renderer will be a new field type in Directus. It will be available as an option when creating or editing a field in a collection, similar to the existing field types 

Interface Configuration: The interface for the HTML Renderer will be configured to accept and display HTML content. When this field type is selected, any HTML content added to the field should be rendered and displayed in the Directus UI. This will likely involve creating a new interface using the Directus extensions SDK 

Rendering HTML Content: The HTML Renderer will use a safe method to parse and display HTML content. This is to prevent potential security issues such as cross-site scripting (XSS) attacks. It could be implemented similarly to the react-directus-html component, which styles and renders HTML content received from Directus 

Accessing Internal Systems: The HTML Renderer will need to access internal systems like the API or the stores. This can be done using the useApi() and useStores() composables exported by the @directus/extensions-sdk package, as shown in the Directus documentation

Handling Relative URLs: The HTML Renderer will need to handle relative URLs that may be included in the HTML content. For example, if an image source URL is relative, the HTML Renderer should replace it with an absolute URL. This could be done similarly to the directusUrl prop used in the react-directus-html component

### Requirements List

Must Have: These are the non-negotiable requirements that the HTML Renderer field type needs to function effectively.
 The field type must safely render HTML content directly within the Directus interface to avoid potential security issues such as cross-site scripting (XSS) attacks [docs.directus.io](https://docs.directus.io/reference/system/fields.html).
 The field type must be available as an option when creating or editing a field in a collection 
 The field type must handle relative URLs that may be included in the HTML content 

Should Have: 
 These are the requirements that are important but not vital to the HTML Renderer field type.
 The field type should be implemented as a Directus extension, similar to the custom displays that can be created using the Directus extensions SDK 
 The HTML Renderer should be able to access internal systems like the API or the stores

Could Have:
 These are the requirements that would enhance the HTML Renderer field type but are not critical for its operation.
 The field type could include options for customizing the display of the rendered HTML, such as changing the font size or color.

Won't Have: 
 These are the requirements that would not be part of the first version of the HTML Renderer field type.
 The field type won't have advanced editing features, like a built-in HTML editor, in its initial release. This feature could be considered for future updates after the basic functionality has been implemented and tested

### Drawbacks

Implementation Cost: Adding a new field type that renders HTML content could increase the complexity of the Directus codebase. It requires careful design and implementation to ensure that the feature is secure, efficient, and compatible with existing Directus features

### Alternatives

nil

### Adoption Strategy

nil

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
